src/platform/plugins/private/links/server/plugin.ts

```diff
@@ -40,10 +41,29 @@ export class LinksServerPlugin implements Plugin<object, object> {
 
     core.savedObjects.registerType<LinksAttributes>(linksSavedObjectType);
 
+    const router = core.http.createRouter();
+
+    router.get(
+      {
+        path: '/api/links',
+        validate: false,
+      },
+      async (context, req, res) => {
+        const [_, { dashboard }] = await core.getStartServices();
+        const { contentClient } = dashboard;
+        const dashboardClient = contentClient!.getForRequest({
+          requestHandlerContext: context,
+          request: req,
+        });
+        const dashboards = await dashboardClient.search({ text: 'web traffic' });
```

Comment:
@nickpeihl Does this handle cross-space searching? We have a similar [need](https://github.com/elastic/kibana/issues/210931) on the client side, and I wanted to make sure that namespaces are exposed on the server side as well

Reply:
It does not. Do you mind creating an issue for the server side request as well?

Which is more important to you, the client or server side? Or are they equal importance?

Reply:
@mgiota I've updated this PR to allow cross-space searching on the server-side via the [`spaces` option](https://github.com/elastic/kibana/pull/209678/files#diff-72b4a5ff836e68c69b0b26da6fd733d56ef77a582d73c68400d804a6004167deR48). 

I added an example [here](https://github.com/elastic/kibana/pull/209678/files#diff-ab22b853956a4223c70b8073d86ee42991b73026525158d1936cedabf4e498adR29-R33).

Reply:
@nickpeihl Thanks a lot for making a change on the server side! Both are equally important. @nreese Do you have an estimate can you could prioritize the [client side](https://github.com/elastic/kibana/issues/210931) as well?

Reply:
> @nreese Do you have an estimate can you could prioritize the https://github.com/elastic/kibana/issues/210931 as well?

No estimate. I will add as a discuss item for our next team meeting

Reply:
Confirmed that spaces is exposed as an option on the client side as well. See https://github.com/elastic/kibana/pull/209678.